ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -317,11 +364,22 @@ boolean isCaughtUp() {
   void shutdown() throws InterruptedException, StoreException, IOException {
     if (running.get()) {
       running.set(false);
-      synchronized (lock) {
-        lock.notify();
+      boolean isPaused = isPaused();
+      if (isPaused) {
+        reentrantLock.lock();
+        paused.set(false);
```

Comment:
You can signal the condition anyway - doesn't have to be that that `pause()` has to been called.